### ISSUE TITLE
Support Japanese charset

### ISSUE THREAD

**Issue by gmarty** [qalle2/findstrings#1]
Hi, I tried to create a table with Japanese characters. Here is a sample:

```
00 0020
01 0
02 1
03 2
04 3
05 4
06 5
07 6
08 7
09 8
0a 9
0b あ
0c い
0d う
0e え
0f お

10 か
11 き
12 く
13 け
14 こ
15 さ
16 し
17 す
18 せ
19 そ
1a た
1b ち
1c つ
1d て
1e と
1f な
```

When running the script, I get the following error message:
```bash
python3 findstrings.py /path/to/binary.gb -t /path/to/table.txt
Syntax error in table file.
```

As a quick fix, I tried to transliterate the characters, which is not ideal, but would have worked as a temporary fix, like this:
```
00 0020
01 0
02 1
03 2
04 3
05 4
06 5
07 6
08 7
09 8
0a 9
0b a
0c i
0d u
0e e
0f o

10 ka
11 ki
12 ku
13 ke
14 ko
15 sa
16 shi
17 su
18 se
19 so
1a ta
1b chi
1c tsu
1d te
1e to
1f na
```

But then I get the following error message:
```
Invalid hexadecimal integer in table file.
```

It seems that it doesn't like mapping to multiple characters either.

I'm not familiar with Python, so I'm not sure how to fix it, but it would be great if you could add support for utf-8 and support more than just ASCII characters.

**Comment by qalle2** [comment]
Thanks for your report. I couldn't replicate your exact problem, but I
fixed related bugs and made the error messages more descriptive.

(Edited)

**Comment by qalle2** [comment]
Nvm, you said you used UTF-8.

**Comment by gmarty** [comment]
I eventually succeeded to make it work by using UTF-8 code points as hex values:
```
10 0020
11 0
12 1
13 2
14 3
15 4
16 5
17 6
18 7
19 8
1a 9
1b 3042
1c 3044
1d 3046
1e 3048
1f 304a

20 304b
21 304d
22 304f
23 3051
24 3053
25 3055
26 3057
27 3059
28 305b
29 305d
2a 305f
2b 3061
2c 3064
2d 3066
2e 3068
2f 306a
```

But the table is extremly tedious to write. Using UTF-8 characters will be way easier.

**Comment by qalle2** [comment]
The program should support non-ASCII characters in the second column of the table files directly as long as you save the file as UTF-8 (without or with byte order mark).
- What error message does the latest version print if you try to use your first table ("10 か" etc.) again?
- What if you use a table file that contains nothing but "10 か"? (with a single space between "10" and "か")

**Comment by gmarty** [comment]
I tried with the minimal test case you suggested and it works.

So I reduced the error and it comes from using the unescaped fullwidth space character (ideographic space | U+3000), for some reason, an entry like this:
```
3f 　
```

causes the `Syntax error in table file.` error message. I supposed it is because white space characters should be encoded as code point.
This works as expected:
```
3f 3000
```

In conclusion, my suggestion would be a more precise error message (indicating the line number where it occurs would help a lot).
